### ISSUE TITLE
ModelStateDictionary causes InvalidOperationException

### ISSUE THREAD

**Issue by hades200082** [domaindrivendev/Swashbuckle.AspNetCore#1208]
## VERSION: ##
5.0.0-rc2

## STEPS TO REPRODUCE: ##
Create a .net core 2.2 WebAPI project. See example code below.

Decorate action methods with `[ProducesResponseType(StatusCodes.Status422UnprocessableEntity, Type = typeof(Microsoft.AspNetCore.Mvc.ModelBinding.ModelStateDictionary))]`

## EXPECTED RESULT: ##
The swagger.json endpoint should work and produce a valid schema.

## ACTUAL RESULT: ##
"Invalid operation: TODO:" possibly tangentially linked to https://github.com/domaindrivendev/Swashbuckle.AspNetCore/commit/6f9e899cfe46e8d99a12e42e378e7c03693dd1c2 but suggests that schema can't be created for ModelStateDictionary

## ADDITIONAL DETAILS ##
Example Startup.cs:

```csharp
public class Startup
{
    public Startup(IConfiguration configuration)
    {
        Configuration = configuration;
    }

    public IConfiguration Configuration { get; }
    public ILogger Log { get; }

    // This method gets called by the runtime. Use this method to add services to the container.
    public void ConfigureServices(IServiceCollection services)
    {
        services.AddApiVersioning(setupAction: options =>
        {
            options.AssumeDefaultVersionWhenUnspecified = true;
            options.DefaultApiVersion = new ApiVersion(majorVersion: 1, minorVersion: 0);
            options.ReportApiVersions = true;
            options.ApiVersionReader = new UrlSegmentApiVersionReader();
        });

        services.AddVersionedApiExplorer(setupAction: options =>
        {
            options.GroupNameFormat = "'v'VV";
            //options.AssumeDefaultVersionWhenUnspecified = true;
        });

        var apiVersionDescriptionProvider =
            services.BuildServiceProvider().GetService<IApiVersionDescriptionProvider>();

        services.AddSwaggerGen(setupAction: options =>
        {
            foreach (var description in apiVersionDescriptionProvider.ApiVersionDescriptions)
            {
                options.SwaggerDoc(name: $"{description.GroupName}", info: new OpenApiInfo
                {
                    Title = "My API",
                    Version = description.ApiVersion.ToString()
                });
            }

            options.DocInclusionPredicate(predicate: (documentName, apiDescription) =>
            {
                var actionApiVersionModel = apiDescription.ActionDescriptor
                    // ReSharper disable once BitwiseOperatorOnEnumWithoutFlags
                    .GetApiVersionModel(mapping: ApiVersionMapping.Explicit | ApiVersionMapping.Implicit);

                if (actionApiVersionModel == null)
                {
                    return true;
                }
                if (actionApiVersionModel.DeclaredApiVersions.Any())
                {
                    return actionApiVersionModel.DeclaredApiVersions.Any(predicate: v =>
                        $"v{v.ToString()}" == documentName);
                }
                return actionApiVersionModel.ImplementedApiVersions.Any(predicate: v =>
                    $"v{v.ToString()}" == documentName);
            });

            options.DescribeAllEnumsAsStrings();

            var xmlCommentsFile = $"{Assembly.GetExecutingAssembly().GetName().Name}.xml";
            var xmlCommentsFullPath = Path.Combine(path1: AppContext.BaseDirectory, path2: xmlCommentsFile);
            options.IncludeXmlComments(filePath: xmlCommentsFullPath);
        });

        services.Configure<ApiBehaviorOptions>(configureOptions: options =>
        {
            options.InvalidModelStateResponseFactory = actionContext =>
            {
                var actionExecutingContext =
                    actionContext as Microsoft.AspNetCore.Mvc.Filters.ActionExecutingContext;

                // if there are model state errors & all keys were correctly
                // found/parsed we're dealing with validation errors
                if (actionContext.ModelState.ErrorCount > 0
                    && actionExecutingContext?.ActionArguments.Count == actionContext.ActionDescriptor.Parameters.Count)
                {
                    return new UnprocessableEntityObjectResult(modelState: actionContext.ModelState);
                }

                // if one of the keys wasn't correctly found / couldn't be parsed
                // we're dealing with null/un-parsable input
                return new BadRequestObjectResult(modelState: actionContext.ModelState);
            };
        });

        services.AddMvc(setupAction: options =>
        {
            options.Filters.Add(item: new ProducesResponseTypeAttribute(statusCode: StatusCodes.Status400BadRequest));
            options.Filters.Add(item: new ProducesResponseTypeAttribute(statusCode: StatusCodes.Status401Unauthorized));
            options.Filters.Add(item: new ProducesResponseTypeAttribute(statusCode: StatusCodes.Status405MethodNotAllowed));
            options.Filters.Add(item: new ProducesResponseTypeAttribute(statusCode: StatusCodes.Status406NotAcceptable));
            options.Filters.Add(item: new ProducesResponseTypeAttribute(statusCode: StatusCodes.Status500InternalServerError));
            options.Filters.Add(item: new ProducesDefaultResponseTypeAttribute());

            options.Filters.Add(item: new ProducesAttribute(contentType: "application/json", "application/xml"));

            options.ReturnHttpNotAcceptable = true;

            options.OutputFormatters.Add(item: new XmlSerializerOutputFormatter());

            var jsonOutputFormatter = options.OutputFormatters
                .OfType<JsonOutputFormatter>().FirstOrDefault();

            if (jsonOutputFormatter != null)
            {
                // remove text/json as it isn't the approved media type
                // for working with JSON at API level
                if (jsonOutputFormatter.SupportedMediaTypes.Contains(item: "text/json"))
                {
                    jsonOutputFormatter.SupportedMediaTypes.Remove(item: "text/json");
                }
            }

        }).SetCompatibilityVersion(version: CompatibilityVersion.Version_2_2);


    }

    // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
    public void Configure(IApplicationBuilder app, IHostingEnvironment env, IApiVersionDescriptionProvider apiVersionDescriptionProvider)
    {
        if (env.IsDevelopment())
        {
            app.UseDeveloperExceptionPage();
        }
        else
        {
            // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
            app.UseHsts();
        }

        app.UseHttpsRedirection();

        app.UseSwagger();
        app.UseSwaggerUI(options =>
        {
            foreach (var description in apiVersionDescriptionProvider.ApiVersionDescriptions)
            {
                options.SwaggerEndpoint($"/swagger/{description.GroupName}/swagger.json", description.GroupName.ToUpperInvariant());
            }
            

            options.RoutePrefix = "SwaggerUI";
            options.DefaultModelExpandDepth(2);
            options.DefaultModelRendering(Swashbuckle.AspNetCore.SwaggerUI.ModelRendering.Model);
            options.DocExpansion(Swashbuckle.AspNetCore.SwaggerUI.DocExpansion.None);
            options.EnableDeepLinking();
            options.DisplayOperationId();
        });

        var latestVersionNumber = apiVersionDescriptionProvider.ApiVersionDescriptions.Max(x => x.ApiVersion);
        var latestVersion =
            apiVersionDescriptionProvider.ApiVersionDescriptions.First(x =>
                x.ApiVersion == latestVersionNumber);
        app.UseReDoc(options =>
        {
            options.SpecUrl($"/swagger/{latestVersion.GroupName}/swagger.json");
            options.DocumentTitle = $"My API {latestVersion.GroupName}";
            options.RoutePrefix = "";
        });
        foreach (var description in apiVersionDescriptionProvider.ApiVersionDescriptions)
        {
            if (description.ApiVersion == latestVersionNumber)
                continue;

            app.UseReDoc(options =>
            {
                options.SpecUrl($"/swagger/{description.GroupName}/swagger.json");
                options.DocumentTitle = $"My API {description.GroupName}";
                options.RoutePrefix = description.GroupName;
            });
        }


        app.UseStaticFiles();

        //app.UseAuthentication();

        app.UseMvc();
    }
}
```

Example controller:

```csharp
[Route("api/v{version:apiVersion}/[controller]")]
[ApiController]
public class ValuesController : ControllerBase
{
    // GET api/values
    [HttpGet]
    [ProducesResponseType(StatusCodes.Status200OK)]
    public ActionResult<IEnumerable<string>> Get()
    {
        return new string[] { "value1", "value2" };
    }

    // GET api/values/5
    [HttpGet("{id}")]
    [ProducesResponseType(StatusCodes.Status404NotFound)]
    [ProducesResponseType(StatusCodes.Status200OK)]
    public ActionResult<string> Get(int id)
    {
        return "value";
    }

    // POST api/values
    [HttpPost]
    [Consumes("application/json")]
    [ProducesResponseType(StatusCodes.Status201Created)]
    [ProducesResponseType(StatusCodes.Status404NotFound)]
    [ProducesResponseType(StatusCodes.Status422UnprocessableEntity,
        Type = typeof(Microsoft.AspNetCore.Mvc.ModelBinding.ModelStateDictionary))]
    public void Post([FromBody] string value)
    {
    }

    // PUT api/values/5
    [HttpPut("{id}")]
    [Consumes("application/json")]
    [ProducesResponseType(StatusCodes.Status200OK)]
    [ProducesResponseType(StatusCodes.Status404NotFound)]
    [ProducesResponseType(StatusCodes.Status422UnprocessableEntity,
        Type = typeof(Microsoft.AspNetCore.Mvc.ModelBinding.ModelStateDictionary))]
    public void Put(int id, [FromBody] string value)
    {
    }

    // PATCH api/values/5
    [HttpPatch("{id}")]
    [Consumes("application/json-patch+json")]
    [ProducesResponseType(StatusCodes.Status200OK)]
    [ProducesResponseType(StatusCodes.Status404NotFound)]
    [ProducesResponseType(StatusCodes.Status422UnprocessableEntity,
        Type = typeof(Microsoft.AspNetCore.Mvc.ModelBinding.ModelStateDictionary))]
    public void Patch(int id, [FromBody] JsonPatchDocument<string> value)
    {
    }

    // DELETE api/values/5
    [HttpDelete("{id}")]
    [ProducesResponseType(StatusCodes.Status204NoContent)]
    [ProducesResponseType(StatusCodes.Status404NotFound)]
    public void Delete(int id)
    {
    }
}
```

**Comment by hades200082** [comment]
Additional info - The sample code works fine if I downgrade to 5.0.0-beta

**Comment by domaindrivendev** [comment]
Not sure when/how this got resolved but doesn't appear to be an issue with current master version, which will be released shortly as `v5.6.0`.

**Comment by domaindrivendev** [comment]
__CORRECTION:__ Using `Type = typeof(ModelStateDictionary)` will still cause problems with `v5.6.0`. However, it's not the correct type to specify. When the framework serializes this type it converts it to `SerializableError` first, and therefore this is the type you should specify with `[ProducesResponseType]`
